### PR TITLE
Added forall, exists aggregators.

### PR DESCRIPTION
**Pull request by cseed**
I want to use them to answer this bioinformatics.SE question:

https://bioinformatics.stackexchange.com/questions/974/selecting-sites-from-vcf-which-have-an-alt-ad-10

```
from hail import *
hc = HailContext()
(hc.import_vcf('test.vcf')
 .filter_variants_expr('gs.exists(g => g.ad[1:].exists(d => d > 10))')
 .count())
```

### CODE REVIEW

File: src/main/scala/is/hail/methods/Aggregators.scala [hunk 1]
```diff
@@ -285,6 +285,48 @@ class FractionAggregator(f: (Any) => Any)
...
+class ExistsAggregator(f: (Any) => Any)
+  extends TypedAggregator[Boolean] {
+
+  var exists: Boolean = false
+
+  def result: Boolean = exists
+
+  def seqOp(x: Any) {
+    val r = f(x)
+    if (r != null && r.asInstanceOf[Boolean])
```
Comment (tpoterba) [inline src/main/scala/is/hail/methods/Aggregators.scala]: shouldn't this short circuit?

Comment (tpoterba) [inline src/main/scala/is/hail/methods/Aggregators.scala]: you shouldn't evaluate f(x) if exists is already true

File: src/main/scala/is/hail/methods/Aggregators.scala [hunk 2]
```diff
@@ -285,6 +285,48 @@ class FractionAggregator(f: (Any) => Any)
...
+
+  def combOp(agg2: this.type) {
+    if (agg2.exists)
+      exists = true
+  }
+
+  def copy() = new ExistsAggregator(f)
+}
+
+class ForallAggregator(f: (Any) => Any)
```
Comment (tpoterba) [inline src/main/scala/is/hail/methods/Aggregators.scala]: can we implement forAll in terms of exists? Or vice versa.

Comment (cseed) [inline src/main/scala/is/hail/methods/Aggregators.scala]: I thought about that but "implementing" one aggregator in terms of another one didn't seem as clear.

File: src/main/scala/is/hail/methods/Aggregators.scala (hunk 1 above)
Comment (cseed) [inline src/main/scala/is/hail/methods/Aggregators.scala]: Good point.  Cleaned up and updated.

File: src/main/scala/is/hail/methods/Aggregators.scala (hunk 2 above)
Comment (tpoterba) [inline src/main/scala/is/hail/methods/Aggregators.scala]: OK, fair enough

**Comment by tpoterba** [comment]
There should also be an example of these in the Expression Language Part 2 tutorial. Would you make an issue for that when you merge?

**Comment by cseed** [comment]
Yep, got it.

**Pull request merged**